### ISSUE TITLE
ANSIBLE0014 blocking ANSIBLE0007 and ANSIBLE0006

### ISSUE THREAD

**Issue by mschuchard** [willthames/ansible-lint#170]
# Issue Type
- Bug report

# Ansible and Ansible Lint details
```
ansible 2.1.0.0
ansible-lint 3.1.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

# Desired Behaviour

```
- name: ANSIBLE0007 and ANSIBLE0014
  command: HOME=/foo /bin/mkdir /path/to/dir creates=/path/to/dir
```

should generate an ANSIBLE0014 warning.

# Actual Behaviour (Bug report only)

```
- name: ANSIBLE0007 and ANSIBLE0014
  command: HOME=/foo /bin/mkdir /path/to/dir creates=/path/to/dir
```

does not generate an ANSIBLE0014 warning.

**Comment by willthames** [comment]
Hmm, that's odd. I'll have to look into this further. I'd have expected tests to pick this up.

I'm not clear what the relationship with #133 is though.

**Comment by willthames** [comment]
The problem here is that I'm only looking for environment variables before the command, not after. 

I assume that, just as when running a command normally, putting environment variables after the command has zero effect. It's also a parsing thing. Some commands take arguments with equals in them. But commands themselves don't contain equals.

```
$ TESTVAR=foo sh -c 'echo $TESTVAR'
foo
$ sh -c 'echo $TESTVAR' TESTVAR=foo
$
```

There is a problem that you can't hit ANSIBLE0007 and ANSIBLE0014 at the same time - if you move `HOME=foo` before the `/bin/mkdir`, the CommandsInsteadOfModulesRule assumes the command name is `HOME=foo`, and so ANSIBLE0007 doesn't get triggered.

**Comment by mschuchard** [comment]
Further testing reveals you are correct. I had assumed during my testing that I had triggered the warning by changing the `command` module to `action`, but it was in fact by moving the ENV part to before the command.

Should ANSIBLE0007 and ANSIBLE0014 be capable of triggering together? I can change the title of this issue if that is what the real prboelm is here.

**Comment by willthames** [comment]
we could strip off the env vars in CommandsInsteadOfModulesRule, but I'm happy enough with the status quo - hopefully people will fix the ANSIBLE0014 breakage and then realise they hit ANSIBLE0007

**Comment by mschuchard** [comment]
I notice ANSIBLE0014 is also blocking ANSIBLE0006, but I guess the same applies there.
Going to update the original issue and close.
